packages/excalidraw/components/App.tsx

```diff
@@ -576,7 +588,52 @@ class App extends React.Component<AppProps, AppState> {
   rc: RoughCanvas;
   unmounted: boolean = false;
   actionManager: ActionManager;
-  device: Device = deviceContextInitialValue;
+  editorInterface: EditorInterface = editorInterfaceContextInitialValue;
+  private stylesPanelMode: StylesPanelMode = deriveStylesPanelMode(
+    editorInterfaceContextInitialValue,
+  );
+  private loadDesktopUIModePreference = () => {
+    if (typeof window === "undefined") {
+      return null;
+    }
+
+    try {
+      const stored = window.localStorage.getItem(DESKTOP_UI_MODE_STORAGE_KEY);
+      if (stored === "compact" || stored === "full") {
+        return stored as EditorInterface["desktopUIMode"];
+      }
+    } catch (error) {
+      // ignore storage access issues (e.g., Safari private mode)
+    }
+
+    return null;
+  };
+
+  private persistDesktopUIMode = (mode: EditorInterface["desktopUIMode"]) => {
+    if (typeof window === "undefined") {
+      return;
+    }
+    try {
+      window.localStorage.setItem(DESKTOP_UI_MODE_STORAGE_KEY, mode);
+    } catch (error) {
+      // ignore storage access issues (e.g., Safari private mode)
+    }
+  };
+
+  public setDesktopUIMode = (mode: EditorInterface["desktopUIMode"]) => {
+    if (mode !== "compact" && mode !== "full") {
+      return;
+    }
+    if (mode === this.editorInterface.desktopUIMode) {
+      return;
+    }
+
+    this.editorInterface = updateObject(this.editorInterface, {
+      desktopUIMode: mode,
+    });
+    this.persistDesktopUIMode(mode);
+    this.reconcileStylesPanelMode(this.editorInterface);
+  };
```

Comment:
this I missed if you mentioned on call, but we shouldn't have this method as public excalidrawAPI either (and also the editor shouldn't concern itself with persistence of user preferences, i.e. be touching localStorage -- even though we still do it with the library and mermaid dialog, but we'll remove those later)